build_resolvers/lib/src/resolver.dart

```diff
@@ -60,14 +68,31 @@ class PerActionResolver implements ReleasableResolver {
       libraries.firstWhere((l) => l.name == libraryName, orElse: () => null);
 
   @override
-  Future<bool> isLibrary(AssetId assetId) => _delegate.isLibrary(assetId);
+  Future<bool> isLibrary(AssetId assetId) =>
+      _resolveIfNecesssary(assetId).then(_delegate.isLibrary);
 
   @override
   Future<LibraryElement> libraryFor(AssetId assetId) =>
-      _delegate.libraryFor(assetId);
+      _resolveIfNecesssary(assetId).then(_delegate.libraryFor);
+
+  Future<AssetId> _resolveIfNecesssary(AssetId id) async {
```

Comment:
In general we avoid returning things just for the purpose of creating a fluid interface (https://dart.dev/guides/language/effective-dart/design#avoid-returning-this-from-methods-just-to-enable-a-fluent-interface). This doesn't fit perfectly with that guideline but I think it fits the spirit of it.

I would probably just return a `Future<void>` here and then use async await at the call sites:

```dart
Future<bool> isLibrary(AssetId assetId) async {
  await _resolveIfNecesssary(assetId);
  return _delegate.isLibrary(assetId);
}
```

I am curious what @natebosch thinks... I don't feel super strongly given this is a private api.

Reply:
I agree with @jakemac53 - let's make this return `Future<void>`